lib/compilers/clang.js

```diff
@@ -38,4 +45,32 @@ export class ClangCompiler extends BaseCompiler {
 
         return super.runCompiler(compiler, options, inputFilename, execOptions);
     }
+
+    async extractDeviceCode(result, filters, outputFilename) {
+        // Check to see if there is any offload code in the assembly file.
+        const offload_regexp = /__CLANG_OFFLOAD_BUNDLE__/;
+        if (!offload_regexp.exec(result.asm) || !this.compiler.deviceExtractor)
```

Comment:
nit: can we reverse these conditions to avoid executing the regex on compilers that don't claim support?